### PR TITLE
Fix sinks washing things at a distance

### PR DESCRIPTION
**Pull request by Arokha**
Sleep replaced with do_after

### CODE REVIEW

File: code/game/objects/structures/watercloset.dm [hunk 1]
```diff
@@ -443,13 +444,12 @@
...
-	sleep(40)
+	if(!do_after(user, 40, src))
+		busy = 0
+		to_chat(usr, "<span class='notice'>You stop washing \the [I].</span>")
+		return
 	busy = 0
 
-	if(user.loc != location) return				//User has moved
-	if(!I) return 								//Item's been destroyed while washing
-	if(user.get_active_hand() != I) return		//Person has switched hands or the item in their hands
```
Comment (MistakeNot4892) [inline code/game/objects/structures/watercloset.dm:451]: This seems like an unnecessary removal.

Comment (Atermonera) [reply code/game/objects/structures/watercloset.dm:451]: Those're all handled by the do_after(), if the state changes between the start and end of it, it'll fail, hitting the to_chat()/return above

Comment (Arokha) [reply code/game/objects/structures/watercloset.dm:451]: ^

Comment (MistakeNot4892) [reply code/game/objects/structures/watercloset.dm:451]: Realized I was thinking of old surgery where you can swap hands, don't mind me.

**Pull request merged**